src/pages/execution/tabLoader.tsx

```diff
@@ -255,6 +265,76 @@ export default function tabLoader(tabBase: TabBase): TabData {
   };
 }
 
+function AlertComponent() {
+  const executionAndTasks = useContext(TileFactoryContext).executionAndTasks;
+  const { severity, setSeverity, setHasAlerts } =
+    useContext(TileFactoryContext);
+  const alerts = useMemo(() => {
+    const allAlerts: AlertItem[] = [];
+
+    rules.forEach((rule) => {
+      const [ruleAlerts, maxSeverity] = rule(executionAndTasks, severity);
+      allAlerts.push(...ruleAlerts);
+      setSeverity(maxSeverity);
+    });
+
+    return allAlerts;
+  }, [executionAndTasks, severity, setSeverity]);
```

Comment:
With this change, no need to depend on `severity` and `setSeverity` here. `alerts` can be a setMemo based only on `executionAndTasks`

Reply:
`setSeverity` can be invoked in a `useEffect` that in turn depends only on `alerts`